### ISSUE TITLE
Uncaught exception: ssl.SSLError: ssl handshake failure

### ISSUE THREAD

**Issue by kseistrup** [TheKysek/MiNode#4]
Spotted in the wild:

``` pytb
Exception in thread Connection to ${IP_ADDRESS_AND_PORT}:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "…/MiNode/src/connection.py", line 78, in run
    self._process_queue()
  File "…/MiNode/src/connection.py", line 154, in _process_queue
    self._send_message(m)
  File "…/MiNode/src/connection.py", line 131, in _send_message
    self.s.sendall(m.to_bytes())
  File "/usr/lib/python3.5/ssl.py", line 891, in sendall
    v = self.send(data[count:])
  File "/usr/lib/python3.5/ssl.py", line 861, in send
    return self._sslobj.write(data)
  File "/usr/lib/python3.5/ssl.py", line 586, in write
    return self._sslobj.write(data)
ssl.SSLError: [SSL: SSL_HANDSHAKE_FAILURE] ssl handshake failure (_ssl.c:1844)
```

**Comment by bmng-dev** [comment]
The failure of ssl handshakes should be expected (which would appear to be the point of this issue) but what is of concern to me is when this particular handshake failure occured - during some arbitrary send rather than do_handshake. The do_handshake loop does not appear to be properly implemented: [SSL: Notes on non-blocking sockets](https://docs.python.org/3/library/ssl.html#notes-on-non-blocking-sockets)

**Comment by kseistrup** [comment]
Additionally, if SSLv3 has been disabled (as it should, IMHO), the following `SSLError` may happen:

``` pytb
Exception in thread Connection to ${IP_ADDRESS_AND_PORT}:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "…/MiNode/src/connection.py", line 62, in run
    data = self.s.recv(1024)
  File "/usr/lib/python3.5/ssl.py", line 914, in recv
    return self.read(buflen)
  File "/usr/lib/python3.5/ssl.py", line 791, in read
    return self._sslobj.read(len, buffer)
  File "/usr/lib/python3.5/ssl.py", line 577, in read
    v = self._sslobj.read(len)
ssl.SSLError: [SSL: SSLV3_ALERT_UNEXPECTED_MESSAGE] sslv3 alert unexpected message (_ssl.c:1977)
```

**Comment by TheKysek** [comment]
Resolved by 00920856cc25571d3d5651b8e8ff9cdb8faa9cd2
